### PR TITLE
bramble: add tmux window naming and session resume on restart

### PR DESCRIPTION
**Pull request by mzhaom**
## Summary
- Tmux windows now use meaningful `repo/worktree:N` names (e.g., `yoloswe/resume-tmux:0`) instead of random two-word names, with per-worktree index allocation
- On exit, active tmux session metadata (window name, ID, runner type) is persisted to the session store
- On restart, `ReconcileTmuxSessions()` checks stored sessions against live tmux windows — re-adopts running ones and marks gone ones as completed
- `TrackTmuxWindow()` now sets `RepoName` on tracked sessions so they persist correctly

## Test plan
- [x] `bazel build //bramble/...` passes
- [x] `scripts/lint.sh` passes
- [x] `bazel test //bramble/...` passes (all 6 targets)
- [x] Unit tests for `GenerateTmuxWindowName(repo, worktree)` format and fallback
- [x] Unit tests for `ReconcileTmuxSessions` (marks gone windows completed, skips non-tmux, skips completed, noop without store/in TUI mode)
- [x] Unit tests for tmux field roundtrip through store save/load, `SessionToStored`, `StoredToSessionInfo`
- [x] Manual: start bramble in tmux, create sessions → verify window names are `repo/worktree:0`, `:1`, etc.
- [ ] Manual: quit bramble → verify `~/.bramble/sessions/` has tmux metadata saved
- [ ] Manual: restart bramble → verify running windows are re-adopted and shown as active

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Touches tmux session lifecycle and persistence (store schema, startup/shutdown reconciliation, and window switching), which can impact whether sessions are incorrectly marked complete or re-adopted. Changes are well-covered by new unit tests but still depend on real tmux behavior and edge cases (renames, missing IDs).
> 
> **Overview**
> Adds tmux session *resume on restart*: `main` reconciles the initial repo’s tmux sessions and discovers other repos with live tmux windows, then `app.Model` auto-opens those repos and re-adopts their sessions on init.
> 
> Updates tmux session persistence and identification by storing `TmuxWindowName`, stable `TmuxWindowID`, and `RunnerType` in the session store; `session.Manager.ReconcileTmuxSessions` re-hydrates still-live sessions and marks stale ones completed, while `Close()` now persists active tmux sessions before shutdown.
> 
> Improves tmux window handling by generating meaningful `repo/worktree:N` names, capturing window IDs atomically at creation, and centralizing window switching via `selectTmuxWindowCmd`; the UI now falls back to tmux window names for session labels and prefers ID-based switching when available.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 47520b413d955da4e31e900d15f5107b67050221. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: bramble/session/manager.go [hunk 1]
```diff
@@ -464,8 +464,139 @@ func (m *Manager) IsInTmuxMode() bool {
...
+			m.mu.Lock()
+			m.sessions[session.ID] = session
+			m.models[session.ID] = sessionmodel.NewSessionModel(1000)
+			m.mu.Unlock()
+
+			m.outputsMu.Lock()
+			m.outputs[session.ID] = make([]OutputLine, 0, 16)
+			m.outputsMu.Unlock()
+
+			m.updateSessionStatus(session, StatusRunning)
```
[the remark this answers is not in the document]
Comment (mzhaom) [reply bramble/session/manager.go]: Fixed: Skip updateSessionStatus for re-adopted sessions to preserve original StartedAt.

File: bramble/session/manager.go [hunk 2]
```diff
@@ -796,16 +939,70 @@ func (m *Manager) monitorTrackedTmuxWindow(session *Session) {
...
+						session.mu.Unlock()
+						m.addOutput(sessionID, OutputLine{
+							Timestamp: time.Now(),
+							Type:      OutputTypeError,
+							Content:   fmt.Sprintf("Session failed: claude exited in tmux window %q. Switch to that window to see the error.", windowName),
+						})
+					} else {
+						// Clean exit (code 0) or couldn't read status — mark completed
+						m.updateSessionStatus(session, StatusCompleted)
+					}
```
[the remark this answers is not in the document]
Comment (mzhaom) [reply bramble/session/manager.go]: Fixed: Unknown exit status now treated as StatusFailed, matching runSession behavior.
